specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/execution.json

```diff
@@ -0,0 +1,1103 @@
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "name": "experimentName",
+            "in": "path",
+            "description": "The experiment name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "definition",
+            "in": "body",
+            "description": "A JSON run definition structure.",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/RunDefinition"
+            }
+          },
+          {
+            "name": "runId",
+            "in": "query",
+            "description": "A run id. If not supplied a run id will be created automatically.",
+            "required": false,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9][\\w-]{0,255}$"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A snapshot run was successfully started.",
+            "schema": {
+              "$ref": "#/definitions/RunStatus"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Start a snapshot run": {
+            "$ref": "./examples/startSnapshotRun.json"
+          }
+        }
+      }
+    },
+    "/execution/v1.0/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/experiments/{experimentName}/cancel": {
+      "post": {
+        "tags": [
+          "Execution"
+        ],
+        "summary": "Cancel a run.",
+        "description": "Cancels a run within an experiment.",
+        "operationId": "Execution_CancelRun",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "name": "experimentName",
+            "in": "path",
+            "description": "The experiment name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "runDetails",
+            "in": "body",
+            "description": "The details of the run to cancel.",
+            "required": false,
```

Comment:
Isn't this required? What happens if I don't specify runDetails?

Reply:
This API shouldn't be documented; Execution_CancelRunWithUri provides the same functionality with a modern contract.

Reply:
Will remove it.

Reply:
done